packages/node/src/integrations/requestdata.ts

```diff
@@ -0,0 +1,127 @@
+} from '../requestdata';
+
+type RequestDataOptions = {
+  /**
+   * Controls what data is pulled from the request and added to the event
+   */
+  include?: {
+    cookies?: boolean;
+    data?: boolean;
+    headers?: boolean;
+    ip?: boolean;
+    query_string?: boolean;
+    url?: boolean;
+    user?: boolean | Array<typeof DEFAULT_USER_INCLUDES[number]>;
+  };
+
+  /** Whether to identify transactions by parameterized path, parameterized path with method, or handler name */
+  transactionNamingScheme?: TransactionNamingScheme;
+
+  /**
+   * Function for adding request data to event. Defaults to `addRequestDataToEvent` from `@sentry/node` for now, but
+   * left injectable so this integration can be moved to `@sentry/core` and used in browser-based SDKs in the future.
+   *
+   * @hidden
+   */
+  addRequestData?: typeof addRequestDataToEvent;
+};
+
+const DEFAULT_OPTIONS = {
+  addRequestData: addRequestDataToEvent,
+  include: {
+    cookies: true,
+    data: true,
+    headers: true,
+    ip: false,
+    query_string: true,
+    url: true,
+    user: DEFAULT_USER_INCLUDES,
+  },
+  transactionNamingScheme: 'methodpath',
+};
+
+/** Add data about a request to an event. Primarily for use in Node-based SDKs, but included in `@sentry/integrations`
+ * so it can be used in cross-platform SDKs like `@sentry/nextjs`. */
+export class RequestData implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'RequestData';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string = RequestData.id;
+
+  private _options: RequestDataOptions;
+
+  /**
+   * @inheritDoc
+   */
+  public constructor(options: RequestDataOptions = {}) {
+    this._options = {
+      ...DEFAULT_OPTIONS,
+      ...options,
+      include: {
+        // @ts-ignore It's mad because `method` isn't a known `include` key. (It's only here and not set by default in
+        // `addRequestDataToEvent` for legacy reasons. TODO (v8): Change that.)
+        method: true,
+        ...DEFAULT_OPTIONS.include,
+        ...options.include,
+      },
+    };
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(addGlobalEventProcessor: (eventProcessor: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    const { include, addRequestData } = this._options;
+
+    addGlobalEventProcessor(event => {
+      const self = getCurrentHub().getIntegration(RequestData);
+      const req = event.sdkProcessingMetadata && event.sdkProcessingMetadata.request;
+
+      // If the globally installed instance of this integration isn't associated with the current hub, `self` will be
+      // undefined
+      if (!self || !req) {
+        return event;
+      }
+
+      // For some reason TS can't figure out that after the constructor runs, `_addReqDataCallback` will always be defined
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      return addRequestData!(event, req, { include: formatIncludeOption(include) });
```

Comment:
Wdyt about removing this non-null-assertion and making turning the type of `_options` into `Required<RequestDataOptions>`? The constructor options are still optional that way. Just want to get rid of as many places where we disable the type checker as we can.

Reply:
I decided to go the other way (make everything required and have the constructor input be a partial) because it seems every-so-slightly more correct to me, but it still gets rid of the non-null assertion. That said, if we're worried about turning off the linter in this situation, maybe we should talk about whether we actually want to lint against `!` at all. After all, we don't lint against `as` typecasts (because we assume that if you're doing it, you're doing it for a good reason), and they're just a different way of saying the same thing.

Reply:
No need to change it now but I would have intentionally made it `Required` since it's pretty unusual to have user-facing APIs that are `Optional`. Doing it the other way would have kept the user-facing type simple to parse mentally.